### PR TITLE
Use gemspec to load C extension

### PR DESCRIPTION
**Pull request by thijsc**
When you add `ext` to the require paths in the gemspec you can use the Rust extension with just `bundle exec` instead of extra command-line params to Ruby. This will also make the gem usable out of the box when installed in another project.

### CODE REVIEW

**Comment by thijsc** [comment]
I've also added the compiled files to the gitignore since I was overwriting them. Does that make sense?

File: .gitignore [hunk 1]
```diff
@@ -20,3 +20,6 @@ tmp
 mkmf.log
 target
 ruby/ext/rust_example/Makefile
+ruby/ext/rust_example/rust_example.o
+ruby/ext/rust_example/rust_example.so
+ruby/ext/rust_example/librust_example.a
```
Comment (steveklabnik) [inline .gitignore]: I specifically left these in so that you can try to repo out even if you don't have Rust.

File: ruby/rust_example.gemspec [hunk 2]
```diff
@@ -16,7 +16,7 @@ Gem::Specification.new do |spec|
   spec.files         = `git ls-files -z`.split("\x0")
   spec.executables   = spec.files.grep(%r{^bin/}) { |f| File.basename(f) }
   spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})
-  spec.require_paths = ["lib"]
+  spec.require_paths = ["lib", "ext"]
```
Comment (steveklabnik) [inline ruby/rust_example.gemspec]: ah ha, I _knew_ I was missing something

**Comment by steveklabnik** [comment]
Thanks for this! a few coments inline

File: .gitignore (hunk 1 above)
Comment (thijsc) [inline .gitignore]: That makes sense. Maybe we could move this to `ruby/ext/rust_example/{arch}` so you can copy them if needed?

Comment (steveklabnik) [inline .gitignore]: This is where they get built by default, though. So you'd need to copy them over.

Really, this repo isn't for serious-grade code, so the fact that they're here just isn't a big deal.

Comment (thijsc) [inline .gitignore]: True. The only reason I deleted them is that I was rebuilding and then accidentally committed the changed files.

**Pull request closed without merging**